Update transaction related bbes (#2940)
ballerina-platform/ballerina-distribution: examples/check-semantics/check_semantics.md

# Check semantics

> `check` semantics is not to simply return on an `error` value.

Reviewer: ```suggestion
`check` semantics does not simply return an `error` value.
```

When `check` gets an `error` value, it fails. The enclosing block decides how to handle the failure.

> Most blocks pass the failure up to the enclosing block. Function definition handles the failure by returning the error.

Reviewer: ```suggestion
Most blocks pass the failure up to the enclosing block. The function definition handles the failure by returning the error.
```

`on fail` can catch the error.

> `fail` statement is like `check`, but it always fails and differs from exceptions in that control flow is explicit.

Reviewer: ```suggestion
The `fail` statement is like `check`, but it always fails. It differs from the exceptions in that the control flow, which are explicit.
```
Please check if the grammatical changes suggested are affecting the technicalities.
